File: 2022/Data_Manager.js
"use strict";
exports.__esModule = true;
exports.filename = exports.List_of_Enemies = exports.game = void 0;
var Definitions_1 = require("./Definitions");
var File_1 = require("./File");
var fs = require("fs");
/*
 * read input
 */
// read all files in inputs folder
var files = fs.readdirSync('./inputs');
var filename = files[5];
exports.filename = filename;
var content = (0, File_1.readFile)('./inputs/' + filename);
var data = content.split("\n");
// parse input
var player_stats = data[0].split(' ');
var Stamina_current = player_stats[0];
var Stamina_max = player_stats[1];
var Turns_max = player_stats[2];
var enemy_amount = player_stats[3];
console.log('Stamina_current: ' + Stamina_current);
console.log('Stamina_max: ' + Stamina_max);
console.log('Turns_max: ' + Turns_max);
console.log('enemy_amount: ' + enemy_amount);
// create player from parsed input
var player = new Definitions_1.Player(parseInt(Stamina_current), parseInt(Stamina_max));
//console.log(player)
var amount_demons = data.length;
var List_of_Enemies = [];
exports.List_of_Enemies = List_of_Enemies;
for (var i = 1; i < amount_demons; i++) {
    var deamon_stats = data[i].split(' ');
    var Stamina_cost = parseInt(deamon_stats[0]);
    var Turn_recover = parseInt(deamon_stats[1]);
    var Stamina_gain = parseInt(deamon_stats[2]);
    var amount_of_turns_gain_fragments = parseInt(deamon_stats[3]);
    //fragments_list_length = len(deamon_stats)
    var fragments_list_length = deamon_stats.length;
    //console.log(fragments_list_length)
    var list_of_fragments = [];
    for (var j = 4; j < fragments_list_length; j++) {
        list_of_fragments.push(parseInt(deamon_stats[j]));
    }
    var demon = new Definitions_1.Demon(i, Stamina_cost, Turn_recover, Stamina_gain, list_of_fragments);
    List_of_Enemies.push(demon);
}
// create game
var game = new Definitions_1.Game(player, parseInt(Turns_max), [], List_of_Enemies, []);
exports.game = game;

File: pythonJail/Data_Manager.py
#class Player:
#  def __init__(self, name, age):
#    self.name = name
#    self.age = age


file = open("inputs/00-example", "r")
data = file.read().split('\n')
print (data)

#print (data[0])

player_stats = data[0].split(' ')

#print (player_stats)

Stamina_current = player_stats[0]
Stamina_max     = player_stats[1]
Turns_max       = player_stats[2]
enemy_amount    = player_stats[3]

print ('Stamina_current: ' + Stamina_current)
print ('Stamina_max: ' + Stamina_max)
print ('Turns_max: ' + Turns_max)
print ('enemy_amount: ' + enemy_amount)

print (len(data))
count = 0
for len in data:
    count += 1
print (count)

#amount_demons = len(data)
amount_demons = count

class deamon:
  def __init__(self, Stamina_cost, Turn_recover,Stamina_gain,amount_of_turns_gain_fragments,fragments_per_turn):
    self.Stamina_cost = Stamina_cost
    self.Stamina_gain = Stamina_gain
    self.amount_of_turns_gain_fragments = amount_of_turns_gain_fragments
    self.amount_of_turns_gain_fragments = amount_of_turns_gain_fragments
    self.fragments_per_turn = fragments_per_turn

List_of_Enemies = []

for i in range(1,amount_demons):
    #print (data[i])
    #print (i)
    deamon_stats = data[i].split(' ')
    Stamina_cost = deamon_stats[0]
    Turn_recover = deamon_stats[1]
    Stamina_gain = deamon_stats[2]
    amount_of_turns_gain_fragments = deamon_stats[3]
    #fragments_list_length = len(deamon_stats)
    #print (fragments_list_length)
    #List_of_Enemies.append(deamon(Stamina_cost, Turn_recover,Stamina_gain,amount_of_turns_gain_fragments))

print(type(List_of_Enemies))
print(type(len(List_of_Enemies)))
x = len(List_of_Enemies)
print(str(x))
